filesystem/fat32/directoryentry.go

```diff
@@ -413,11 +416,47 @@ func convertLfnSfn(name string) (string, string, bool, bool) {
 		isLFN = true
 	}
 
+	truncate := func(name string, n int) string {
+		return name[0:6] + "~" + strconv.Itoa(n)
+	}
```

Comment:
I am not madly in love with inline anonymous functions here, as they will get declared every time we call `convertLfnSfn`, and they are harder to understand and test.

Can we pull these three out into named functions, and give them more representative names?